### PR TITLE
patient's unit test: done; user u…

### PR DESCRIPTION
**Pull request by tnguyen10**
PATIENT UNIT TEST
- test getProblem, getProblems, addProblem, removeProblem, setProblem

USER UNIT TEST
- test setters and getters
- test non numerical value for setUserId
- test non numerical value for constructor

OTHERS
- 1 constructor for user, patient, provider

### CODE REVIEW

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java [hunk 1]
```diff
@@ -3,15 +3,15 @@
 import android.location.Location;
 
 import java.util.ArrayList;
+import java.util.Date;
 
 public class Problem extends Entry implements Refreshable {
     protected String description;
     protected ArrayList<Record> records = new ArrayList<>();
```
Comment (modelarious) [inline app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: Can we keep the 0-argument constructor available for testing?

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
 
-    Problem() {
+    public Problem(String createdByUserID, String title) {
         super();
```
Comment (modelarious) [inline app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: Needs to call super(createdByUserID, title);

Comment (tnguyen10) [reply app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: Should be fixed

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java (hunk 1 above)
Comment (tnguyen10) [reply app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: Yup

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java [hunk 3]
```diff
@@ -3,15 +3,16 @@
...
 import java.util.ArrayList;
+import java.util.Date;
 
 public class Problem extends Entry implements Refreshable {
     protected String description;
     protected ArrayList<Record> records = new ArrayList<>();
 
-    Problem() {
-        super();
+    public Problem();
```
Comment (modelarious) [inline app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: Requires an implementation:
public Problem() {}

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java [hunk 4]
```diff
@@ -3,15 +3,16 @@
...
+    public Problem(String createdByUserID, String title) {
```
Comment (modelarious) [inline app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: I believe this would throw a NonNumericUserIDException.

**Comment by modelarious** [comment]
The changes look pretty good.  Could you try compiling and running the tests?  Should reveal a few unhandled exceptions from the changes you're pushing.  Want to make sure every time that we commit to the master branch that the project is in a working state.

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java (hunk 3 above)
Comment (tnguyen10) [reply app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: fixed

File: app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java (hunk 4 above)
Comment (tnguyen10) [reply app/src/main/java/com/cmput301f18t20/medicalphotorecord/Problem.java]: fixed

**Comment by tnguyen10** [comment]
the try catch block for set user id are as messy as constructor. I don't think we need to implement nonnumericallyexception since the requirement doesn't not specify so?

**Comment by modelarious** [comment]
I've addressed the exceptions and handling them in test cases.

**Pull request merged**